api/src/main/java/io/opentelemetry/trace/Span.java

```diff
@@ -23,6 +23,26 @@
 @ThreadSafe
 public interface Span {
 
+  /**
+   * Returns an invalid {@link Span}. An invalid {@link Span} is used when tracing is disabled,
+   * usually because there is no OpenTelemetry SDK installed.
+   */
+  static Span getInvalid() {
+    return DefaultSpan.INVALID;
+  }
+
+  /**
+   * Returns a {@link Span} that only wraps the provided {@link SpanContext} but has no
+   * functionality. It will not be exported and all operations are no-op, but it can be used to
+   * propagate the {@link SpanContext} downstream.
+   */
+  static Span wrapWithNoop(SpanContext spanContext) {
```

Comment:
@Oberon00 Based on the conversation in https://github.com/open-telemetry/opentelemetry-specification/issues/1086 this should be something like `wrapForPropagation`, we're wrapping this `SpanContext` into a `PropagatedSpan`. This seems like odd naming to me though, as there is nothing about propagation happening here really, doubly-so because propagation is still more commonly used for remote propagation than in-process propagation I think. That's not great here since we don't do remote propagation of the invalid span.

Reply:
>  as there is nothing about propagation happening here really

If one considers making a span a local parent "propagation" (which I think one should), then it makes sense. Because what other uses would this returned object have?

Reply:
> Because what other uses would this returned object have?

I dunno maybe someone will have something creative :) While it's indirect usage of this API, one case where it will be used is when extracting the `SpanContext` from a link - the result of this method will be taken out of the context, the link will be made, and the result will be dropped without ever being propagated. So this is where the disconnect with the naming comes from - since it's completely up to the caller on whether it's propagated or not, so I find it pretty confusing. There's the other point where there's no such thing as a non-propagated span - all spans can be propagated, so why do we have the propagated span called out in particular? I'm sure someone is going to call `Span.getPropagated(tracer.startSpan().getContext())` assuming that's how you're supposed to propagate your span ;) `Span.getNoop(tracer.startSpan().getContext())`, `Span.getNonRecording(...)` seems more obviously wrong.

Reply:
I don't love this name, but I 100% agree with the sentiment. Making it super clear what the capabilities (and lack thereof) will be of the returned Span seems like a very important signal to the user.